crates/bevy_image/src/image_loader.rs

```diff
@@ -179,7 +191,31 @@ impl AssetLoader for ImageLoader {
         .map_err(|err| FileTextureError {
             error: err,
             path: format!("{}", load_context.path().display()),
-        })?)
+        })?;
+
+        if let Some(view_dimension) = &settings.view_dimension {
+            match view_dimension {
+                ImageTextureViewDimension::D2Array(layers) => {
+                    image.reinterpret_stacked_2d_as_array(*layers)?;
+                }
+                ImageTextureViewDimension::Cube => {
+                    image.reinterpret_stacked_2d_as_array(image.height() / image.width())?;
+                }
+                ImageTextureViewDimension::CubeArray(layers) => {
+                    image.reinterpret_stacked_2d_as_array(
+                        image.height() / image.width() * *layers,
+                    )?;
```

Comment:
I have no idea about Cube or CubeArray textures. Not sure if this was something specific to the skybox example or if Cube will always be height/width.